### PR TITLE
Evm-C updates

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: include/evm.h [hunk 1]
```diff
@@ -154,12 +154,12 @@ union evm_variant {
 /// ::EVM_ORIGIN          |                      | evm_variant::address
 /// ::EVM_COINBASE        |                      | evm_variant::address
 /// ::EVM_DIFFICULTY      |                      | evm_variant::uint256
-/// ::EVM_GAS_LIMIT       |                      | evm_variant::int64
+/// ::EVM_GAS_LIMIT       |                      | evm_variant::uint256? or uint64?
```
Comment (chfast) [inline include/evm.h]: Definitely `int64`. Ethereum lowered the max block gas limit to 2^63-1.

File: include/evm.h [hunk 2]
```diff
@@ -319,12 +336,15 @@ EXPORT void evm_destroy_result(struct evm_result);
 /// @{
 
 
-EXPORT bool evmjit_is_code_ready(evm_instance* instance, evm_mode mode,
-                                 evm_hash256 code_hash);
+EXPORT bool evmjit_is_code_ready(struct evm_instance* instance,
+                                 struct evm_mode mode,
```
Comment (chfast) [inline include/evm.h]: `struct` is wrong here.

File: include/evm.h [hunk 3]
```diff
@@ -319,12 +336,15 @@ EXPORT void evm_destroy_result(struct evm_result);
...
+                                 struct evm_hash256 code_hash);
 
-EXPORT void evmjit_compile(evm_instance* instance, evm_mode mode,
-                           uint8_t const* code, size_t code_size,
-                           evm_hash256 code_hash);
+EXPORT void evmjit_compile(struct evm_instance* instance,
+                           struct evm_mode mode,
```
Comment (chfast) [inline include/evm.h]: `struct` is wrong here.

File: include/evm.h [hunk 4]
```diff
@@ -174,7 +174,24 @@ enum evm_update_key {
...
 
 
-/// Callback function for modifying a contract state.
+/// Update callback function.
+///
+/// This callback function is used by the EVM to modify contract state in the
+/// host application.
+/// @param env  Pointer to execution environment managed by the host
+///             application.
+/// @param key  The kind of the query. See evm_update_key and details below.
```
Comment (chfast) [inline include/evm.h]: Replace "query" with "update" here and in the following lines.

File: include/evm.h [hunk 5]
```diff
@@ -154,12 +154,12 @@ union evm_variant {
...
 /// ::EVM_COINBASE        |                      | evm_variant::address
 /// ::EVM_DIFFICULTY      |                      | evm_variant::uint256
-/// ::EVM_GAS_LIMIT       |                      | evm_variant::int64
+/// ::EVM_GAS_LIMIT       |                      | evm_variant::uint256? or uint64?
 /// ::EVM_NUMBER          |                      | evm_variant::int64?
 /// ::EVM_TIMESTAMP       |                      | evm_variant::int64?
 /// ::EVM_CODE_BY_ADDRESS | evm_variant::address | evm_variant::bytes
 /// ::EVM_BALANCE         | evm_variant::address | evm_variant::uint256
-/// ::EVM_BLOCKHASH       | evm_variant::int64   | evm_variant::uint256
+/// ::EVM_BLOCKHASH       | evm_variant::int64   | evm_variant::hash256
```
Comment (chfast) [inline include/evm.h]: Correct. Thanks.

File: include/evm.h (hunk 2 above)
Comment (axic) [inline include/evm.h]: Why? `evm_instance` is defined as a struct, but not typedef'd. Everywhere else it is referred to as a struct. Without this it won't compile in pure C. It does with a c++11 mode.

File: include/evm.h (hunk 1 above)
Comment (axic) [inline include/evm.h]: `cpp-ethereum` still uses `uint256`.

I agree with the EIP of using 63 bits, but as it goes with every EIP, it wasn't entirely decided yet, right? :)

File: include/evm.h (hunk 2 above)
Comment (axic) [inline include/evm.h]: Ah you meant for the `enum`, that's right. Will fix.

Comment (chfast) [inline include/evm.h]: I mean `struct evm_mode mode` -> `evm_mode mode` because it is an enum.

File: include/evm.h (hunk 1 above)
Comment (chfast) [inline include/evm.h]: VMs in cpp-ethereum (including evmjit) are using `int64` for some time. EIP is decided. Just the higher lever API has not been changed yet.

Comment (chfast) [inline include/evm.h]: Ah, I understand now what you mean. Looks like bug in current implementation, will try to correct it. So place `uint256` here.

**Comment by chfast** [comment]
Can you rebase with latest develop branch? That should fix werker builds.

**Pull request merged**